pkg/controller/factory/base_controller.go

```diff
@@ -62,6 +65,39 @@ func (s *scheduledJob) Run() {
 	s.queue.Add(DefaultQueueKey)
 }
 
+func waitForPreconditions(ctx context.Context, controllerName string, preconditions ...PreconditionFn) error {
+	if len(preconditions) == 0 {
+		return nil
+	}
+
+	var (
+		wg       sync.WaitGroup
+		errs     []error
+		errsLock sync.Mutex
+	)
+
+	wg.Add(len(preconditions))
+	for _, preconditionFunc := range preconditions {
+		go func(preconditionCtx context.Context, fn PreconditionFn) {
+			defer wg.Done()
+			err := wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
```

Comment:
I still don't like this. If I put a client call in here, this will hotloop (modulo throttling).

Reply:
the 100ms should prevent hotloop, no? also client will throttle as well...